File: LeetCode/00-legacy-warmups/search-insert-position/solutions/main.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */

const searchInsert = (nums, target) => {
  let left = 0;
  let right = nums.length - 1;

  while (left <= right) {
    root = Math.floor((left + right) / 2);

    if (nums[root] === target) return root;
    else if (target > nums[root]) left = root + 1;
    else if (target < nums[root]) right = root - 1;
  }

  return left;
};

File: LeetCode/00-legacy-warmups/search-insert-position/solutions/main.go
func searchInsert(nums []int, target int) int {
    
  var start int = 0
  var end int = len(nums)
  var mid int = len(nums) / 2

  for start < mid {
    if target == nums[mid] {
      return mid
    }

    if target > nums[mid] {
      start = mid
    }

    if target < nums[mid] {
      end = mid
    }

    mid = (start + end) / 2
  }

  if target == nums[mid] || target < nums[mid] {
    return mid
  }

  return end
}